src/services/audius-api-client/types.ts

```diff
@@ -97,6 +100,43 @@ export type APITrack = {
   is_delete: boolean
   cover_art: Nullable<string>
   play_count: number
+  is: string
+}
+
+export type APIPlaylistAddedTimestamp = {
+  timestamp: string
+  track_id: OpaqueID
+}
+
+export type APIPlaylist = {
+  artwork: CoverArtSizes
+  description: Nullable<string>
+  id: OpaqueID
+  is_album: boolean
+  playlist_name: string
+  repost_count: number
+  favorite_count: number
+  total_play_count: number
+  user_id: OpaqueID
+  user: APIUser
+  created_at: string
+  updated_at: string
+  followee_reposts: APIRepost[]
+  followee_favorites: APIFavorite[]
+  has_current_user_reposted: boolean
+  has_current_user_saved: boolean
+  is_delete: boolean
+  is_private: boolean
+  added_timestamps: APIPlaylistAddedTimestamp[]
+  tracks: APITrack[]
+}
+
+export type APIItemType = 'track' | 'playlist'
+
+export type APIActivity = {
+  timestamp: string
+  item_type: APIItemType
```

Comment:
I think we want to use our old friend discriminated union for this? We have the type ItemType but we can't use it to narrow the type rn - I think we want something like 
```
APIActivity = { timestamp: string } & ({ item_type: 'track'; item: APITrack } | { item_type: 'playlist'; item: APIPlaylist })
```